website/static/js/recentActivityWidget.js

```diff
@@ -0,0 +1,345 @@
+var $ = require('jquery');
+var m = require('mithril');
+var moment = require('moment');
+var $osf = require('js/osfHelpers');
+var LogText = require('js/logTextParser');
+var pips = require('sliderPips');
+
+var xhrconfig = function (xhr) {
+    xhr.withCredentials = true;
+    xhr.setRequestHeader('Content-Type', 'application/vnd.api+json');
+    xhr.setRequestHeader('Accept', 'application/vnd.api+json; ext=bulk');
+
+};
+
+var LogWrap = {
+    controller: function(args){
+        var self = this;
+        self.wrapper = args.wrapper;
+        self.activityLogs = m.prop();
+        self.eventFilter = false; //holds the currently selected action category filter to include in request
+        self.dateEnd = moment.utc();
+        self.dateBegin = moment.utc();
+        self.today = moment.utc();
+        self.sixMonthsAgo = moment.utc().subtract(6, 'months');
+        self.page = 1; //page currently shown on log container, related to queried page by ` page : ((self.page/2) | 0) + 1 `
+        self.cache = []; //to store already requested with the current parameters
+        self.loading = false; //keeps track of whether the log container should show the loading icon
+        self.div = 8.64e+7; //milliseconds in day, for conversion with the slider bar
+        self.monthDiv = 2629746000;  //milliseconds in a month
+        self.canvasHeight = 40;
+        self.totalEvents = 0; //initialization of count of all logs returned (total)
+        self.eventNumbers = {}; //initialization of dict for aggregate counts (number of 'file' events, etc)
+        self.commentEvents = 0;
+        self.fileEvents = 0;
+        self.wikiEvents = 0;
+        self.nodeEvents = 0;
+        self.otherEvents = 0;
+        self.errorLoading = false;
+        self.select = function(selector) {
+            return $('#'+self.wrapper).find(selector);
+        };
+
+        self.getLogs = function(init, reset, update) {
+            if (!(init || reset || update)  && self.cache[self.page - 1]){
+                self.activityLogs(self.cache[self.page - 1]);
+                if (!self.cache[self.page] && self.page < self.lastPage){
+                    self.page = self.page + 1;
+                    self.getLogs(false, false, true);
+                    self.page = self.page - 1;
+                }
+                return;
+            }
+            var query = {
+                'embed': ['nodes', 'user', 'linked_node', 'template_node'],
+                'page': ((self.page/2) | 0) + 1,
```

Comment:
No, actually this is not an ` or `, it is a bitwise or, to allow for division of integer rounded down:
```
((7/3) || 0)
 = 2.3333333333333335
((7/3) | 0)
 = 2
```
There this is used so that the requested page 3, for example, corresponds to index (` self.page `) 6 and 7.